File: java/82 NestedClass.java
package code;
import java.util.Arrays;
class ArrayOperations{
	int nums[];
	
	public ArrayOperations(int[] nums) {
		this.nums = Arrays.copyOf(nums, nums.length);
	}

	class Statistics {
		double mean() {
			double sum=0;
			for(int num : nums) {
				sum+=num;
			}
			return sum/nums.length;
			
		}
		double median() {
			int[] sortedNums = Arrays.copyOf(nums, nums.length);
            Arrays.sort(sortedNums);

            int middle = sortedNums.length / 2;

            if (sortedNums.length % 2 == 1) {
                return sortedNums[middle];
            } else {
                return (sortedNums[middle - 1] + sortedNums[middle]) / 2.0;
            }
			
		}
	}
}
public class Test {
	public static void main(String[] args) {
		ArrayOperations a1=new ArrayOperations(new int[]{1,3,5,9});
		ArrayOperations.Statistics  s1=a1.new Statistics ();
		System.out.println(s1.mean()+" "+s1.median());
	}
}

File: python/82 NestedClass.py
class ArrayOperations:
    def __init__(self,nums:list):
        self.nums=list(nums)

    class Operation:
        def __init__(self, data_list: list):
            self.data = data_list

        def mean(self):
            return sum(self.data) / len(self.data)
        
        def median(self):
            sorted_data = sorted(self.data)
            n = len(sorted_data)

            if n % 2 == 1:
                return float(sorted_data[n // 2])
            else:
                mid1 = sorted_data[n // 2 - 1]
                mid2 = sorted_data[n // 2]
                return (mid1 + mid2) / 2.0


arr_ops1 = ArrayOperations([1, 5, 2, 8, 3])
op1 = arr_ops1.Operation(arr_ops1.nums)
print(f"Mean: {op1.mean()}")
print(f"Median: {op1.median()}")
